apps/wallet/src/ui/app/pages/home/nfts/index.tsx

```diff
@@ -1,45 +1,51 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { Coin } from '@mysten/sui.js';
+import { useMemo } from 'react';
 import { Link } from 'react-router-dom';
 
 import Alert from '_components/alert';
 import { ErrorBoundary } from '_components/error-boundary';
 import Loading from '_components/loading';
-import NFTdisplay from '_components/nft-display';
-import { useAppSelector, useObjectsState } from '_hooks';
-import { accountNftsSelector } from '_redux/slices/account';
+import { NFTDisplayCard } from '_components/nft-display';
+import { useAppSelector, useObjectsOwnedByAddress } from '_hooks';
 import PageTitle from '_src/ui/app/shared/PageTitle';
 
 function NftsPage() {
-    const nfts = useAppSelector(accountNftsSelector);
-    const { error, loading, showError } = useObjectsState();
-    const isNftsFound = nfts.length > 0;
+    const accountAddress = useAppSelector(({ account }) => account.address);
+    const { data, isLoading, error } = useObjectsOwnedByAddress(accountAddress);
+    const nfts = useMemo(
+        () => data?.filter((obj) => !Coin.isCoin(obj)),
+        [data]
+    );
+
     return (
         <div className="flex flex-col flex-nowrap items-center gap-4 flex-1">
             <PageTitle title="NFTs" />
-            <Loading loading={loading}>
-                {showError && error ? (
+            <Loading loading={isLoading}>
+                {error instanceof Error ? (
```

Comment:
Let's just check if error is truthy here and do something like `(error as Error).message` in the alert